### PR TITLE
Fix type mismatch in Gradle

### PR DESCRIPTION
**Pull request by blasten**
Fixes:
```
FAILURE: Build failed with an exception.
* Where:
Script '/tmp/build/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 458
* What went wrong:
A problem occurred configuring project ':connectivity'.
> Cannot convert the provided notation to an object of type Dependency: /tmp/build/src/out/android_release/flutter.jar.
```

### CODE REVIEW

**Comment by mkustermann** [comment]
It would be great if this fix could land soon (our head-head-head setup is broken until it lands).

@tvolkert  Who would be a good reviewer for gradle changes?

**Review by tvolkert — APPROVED** [review]
Test?

**Comment by blasten** [comment]
We have a build test on Golem that exercises this path. For now, I created https://github.com/flutter/flutter/issues/39762.

**Comment by blasten** [comment]
Tested the changes manually and confirmed that it fixes the issue. I'm going to merge on red on unblock the Golem build.

**Pull request merged**
